tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -35,13 +35,26 @@
 import xml.etree.ElementTree as ET
 import os
 import sys
+import re
+
 import build_cleaner
 
-_ROOT = os.path.abspath(os.path.join(os.path.dirname(sys.argv[0]), '../..'))
-os.chdir(_ROOT)
+PROJECT_ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..",
+                            "..")
+os.chdir(PROJECT_ROOT)
+
+EXTERNAL_LIB_LABEL_PATH_MAP = dict(
+    envoy_api="third_party/envoy-api/",
+    com_google_googleapis="third_party/googleapis/",
+    com_github_cncf_udpa="third_party/udpa/",
+    com_envoyproxy_protoc_gen_validate="third_party/protoc-gen-validate/",
+    com_google_protobuf="third_party/protobuf",
+    opencensus_proto="third_party/opencensus-proto/src/")
+RE_SUPPORTED_LIB = re.compile(
```

Comment:
RE_SUPPORTED_LIB  name is not really descriptive. It provides no explanation of what it's useful for beyond "it's going to match some libraries we're later going to need" which is useless for getting any kind of insight.